osquery/filesystem/fileops.h

```diff
@@ -0,0 +1,211 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <fcntl.h>
+#include <string>
+#include <vector>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#ifdef WIN32
+#define NOMINMAX
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+#include <boost/optional.hpp>
+
+namespace osquery {
+
+#ifdef WIN32
+using mode_t = int;
+using ssize_t = SSIZE_T;
+using PlatformHandle = HANDLE;
+using PlatformTimeType = FILETIME;
+
+/// Windows does not define these, X_OK on Windows just ensures that the
+/// file is readable.
+#define F_OK 0
+#define R_OK 4
+#define W_OK 2
+#define X_OK R_OK
+
+const std::string kFallbackHomeDirectory = "\\ProgramData\\osquery";
```

Comment:
Let's move these out of the header and into an implementation file. Feel free to declare them `extern` if it's used somewhere else. Though, I don't see that in this diff and cannot find `kFallbackHomeDirectory` in the source.